pkg/virt-controller/watch/vm/vm.go

```diff
@@ -863,17 +870,29 @@ func (c *Controller) handleVolumeUpdateRequest(vm *virtv1.VirtualMachine, vmi *v
 	case *vm.Spec.UpdateVolumesStrategy == virtv1.UpdateVolumesStrategyMigration:
 		// Validate if the update volumes can be migrated
 		if err := volumemig.ValidateVolumes(vmi, vm, c.dataVolumeStore, c.pvcStore); err != nil {
-			log.Log.Object(vm).Errorf("cannot migrate the VM. Volumes are invalid: %v", err)
+			if errors.Is(err, storagetypes.ErrPVCNotFound) || errors.Is(err, storagetypes.ErrDVNotFound) {
```

Comment:
Looks good but this function is getting harder to read due to the branching logic. Not a must but it might be worth considering moving this volume-validation logic to a separate function (like `handleVolumeValidation`?). wdyt?

Reply:
Added, thanks!